File: library/src/main/java/com/jianyuyouhun/mobile/fastgather/library/utils/Logger.java
package com.jianyuyouhun.mobile.fastgather.library.utils;

import android.util.Log;

import com.jianyuyouhun.mobile.fastgather.library.app.JApp;

/**
 * 日志打印
 * Created by wangyu on 2018/3/27.
 */

public class Logger {
    public static void i(String tag, String msg) {
        if (JApp.isDebug()) {
            int max_str_length = 2001 - tag.length();
            //大于4000时
            while (msg.length() > max_str_length) {
                Log.i(tag, msg.substring(0, max_str_length));
                msg = msg.substring(max_str_length);
            }
            //剩余部分
            Log.i(tag, msg);
        }
    }

    public static void e(String tag, String msg) {
        if (JApp.isDebug()) {
            int max_str_length = 2001 - tag.length();
            //大于4000时
            while (msg.length() > max_str_length) {
                Log.e(tag, msg.substring(0, max_str_length));
                msg = msg.substring(max_str_length);
            }
            //剩余部分
            Log.e(tag, msg);
        }
    }

    public static void d(String tag, String msg) {
        if (JApp.isDebug()) {
            int max_str_length = 2001 - tag.length();
            //大于4000时
            while (msg.length() > max_str_length) {
                Log.d(tag, msg.substring(0, max_str_length));
                msg = msg.substring(max_str_length);
            }
            //剩余部分
            Log.d(tag, msg);
        }
    }

    public static void w(String tag, String msg) {
        if (JApp.isDebug()) {
            int max_str_length = 2001 - tag.length();
            //大于4000时
            while (msg.length() > max_str_length) {
                Log.w(tag, msg.substring(0, max_str_length));
                msg = msg.substring(max_str_length);
            }
            //剩余部分
            Log.w(tag, msg);
        }
    }
}

File: library/src/main/java/com/jianyuyouhun/mobile/fastgather/library/utils/kt/Logger.kt
package com.jianyuyouhun.mobile.fastgather.library.utils.kt

import android.util.Log
import com.jianyuyouhun.mobile.fastgather.library.app.AbstractJApp

/**
 * 日志打印
 * Created by wangyu on 2017/7/25.
 */
private inline fun doLog(log: () -> Unit) {
    if (AbstractJApp.isDebug()) log()
}

fun lgE(tag: String, msg: String) = doLog { Log.e(tag, msg) }

fun lgI(tag: String, msg: String) = doLog { Log.i(tag, msg) }

fun lgD(tag: String, msg: String) = doLog { Log.d(tag, msg) }

fun lgW(tag: String, msg: String) = doLog { Log.w(tag, msg) }
